### ISSUE TITLE
there is wrong in class 'GatedShapeConv'

### ISSUE THREAD

**Issue by lpengc** [ben-davidson-6/Gated-SCNN#1]
`class GatedShapeConv(tf.keras.layers.Layer):
    def __init__(self, **kwargs):
        super(GatedShapeConv, self).__init__(**kwargs)
        self.conv_1 = None
        self.gated_conv = GateConv()

    def build(self, input_shape):
        feature_channels = input_shape[0][-1]
        self.conv_1 = tf.keras.layers.Conv2D(feature_channels, 1)

    def call(self, x, **kwargs):
        feature_map, shape_map = x
        features = tf.concat([feature_map, shape_map], axis=-1)
        alpha = self.gated_conv(features)
        gated = feature_map*(alpha + 1.)
        return self.conv_1(gated)`

the 'feature_map' in 'gated = feature_map*(alpha + 1.)' should be shape_map in original paper

**Comment by ben-davidson-6** [comment]
I cannot see the discrepency, could you point me to the part of the paper which says this. Following the implementation [here](https://github.com/nv-tlabs/GSCNN/blob/master/network/gscnn.py) written by the authors I think the implementations match up. They do

```python
cs = self.gate1(cs, s3)

# with forward pass of gate1 defined like
def forward(self, input_features, gating_features):
        """
        :param input_features:  [NxCxHxW]  featuers comming from the shape branch (canny branch).
        :param gating_features: [Nx1xHxW] features comming from the texture branch (resnet). Only one channel feature map.
        :return:
        """
        alphas = self._gate_conv(torch.cat([input_features, gating_features], dim=1))

        input_features = (input_features * (alphas + 1)) 
        return F.conv2d(input_features, self.weight, self.bias, self.stride,
                        self.padding, self.dilation, self.groups)
```
